### ISSUE TITLE
no-unbound-method warns when deleting property

### ISSUE THREAD

**Issue by tvuotila** [palantir/tslint#3801]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const a = {
  getText() {
    return 'text';
  }
};

delete a.getText;
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-unbound-method": true
  }
}
```

#### Actual behavior

Error `Avoid referencing unbound methods which may cause unintentional scoping of 'this'.`

#### Expected behavior
No error. I am deleting a property. There is no risk on deleting unbound method compared to other kind of properties.

**Comment by piotrgajow** [comment]
Hi, I think I could fix this issue by adding `allow-delete` parameter similar to what I have done previously for `typeof` operator. What do you think about such solution?

**Comment by JoshuaKGoldberg** [comment]
@piotrgajow that sounds great, yes please! 😄

**Comment by piotrgajow** [comment]
Ok, I have made PR for this change.

**Comment by JoshuaKGoldberg** [comment]
Fxed by #4548! ✨
